include/openenclave/bits/edl/syscall_types.h

```diff
@@ -4,11 +4,39 @@
 #ifndef _OE_EDL_SYSCALL_TYPES_H
 #define _OE_EDL_SYSCALL_TYPES_H
 
+#include <openenclave/bits/defs.h>
+
 /* DISCLAIMER:
  * This header is published with no guarantees of stability and is not part
  * of the Open Enclave public API surface. It is only intended to be used
  * internally by the OE runtime. */
 
 typedef int64_t oe_host_fd_t;
 
+typedef union _oe_epoll_data_t {
+    void* ptr;
+    int fd;
+    uint32_t u32;
+    uint64_t u64;
+} oe_epoll_data_t;
+
+/*
+ * The epoll events are packed in x86_64 but not on other architectures. This is
+ * probably because the x86 has no problem with unaligned accesses where as
+ * unaligned accesses are usually very inefficient, even if they are allowed, in
+ * other architectures. We maintain consistency with MUSL's and the linux
+ * kernel's handling on that architecture.
+ */
+#if !defined(__aarch64__)
+OE_PACK_BEGIN
+#endif
+struct oe_epoll_event
```

Comment:
Do these structs need to be packed at the boundary level? One could use unpacked EDL defined structs to marshal between host and enclave and transfer contents to packed structs only when needed.

Reply:
You're correct, they don't need to be packed at the boundary level.

For epoll in particular, whenever multiple events are passed, they're kept in contiguous memory so they can easily be memcpy'd all at the same time. The code to do the marshalling/unmarshalling is really ugly if you have to do it manually and it doesn't really provide any benefit in this case.